core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -33,12 +33,87 @@ import fs2.internal._
 import Resource.ExitCase
 import Pull._
 
-/** A `p: Pull[F,O,R]` reads values from one or more streams, returns a
-  * result of type `R`, and produces a `Stream[F,O]` when calling `p.stream`.
+/** A pure abstraction representing the intention to perform a sequence of actions
+  * in an effect type `F`, which emits any number of values of an output type `O`.
+  * This sequence may end up in several ways:
+  * a. it may not terminate and keep unrolling new items;
+  * b. it may be aborted by a `Throwable` error;
+  * c. be interrupted by a signal, which may include an error,
+  * d. terminate successfully and returning one single carry-over result of type `R`.
   *
-  * Any resources acquired by `p` are freed following the call to `stream`.
+  * Much like the `IO` values fromo the `cats-effect` library,
+  * `Pull` values are pure, immutable values. Thus, they preserve referential
+  * transparency and are usable in functional programming.
+  * A `Pull` is a data structure that represents just a description
+  * of a sequence of potentially side effectful computation.
   *
-  * Laws:
+  * A pull emits the output values not one-at-a-time, but one-chunk at a time.
+  * A pull only emits non-empty chunks.
+  *
+  * === Pulls variations
+  *
+  * There are some special types of Pull, which are reflected on the arguments of its type parameters:
+  *
+  * A '''pure''' pull is one that performs no effectful actions.
+  * The type of a pure pull has the `F` effect type  bound to `Nothing`.
+  * _Note_ that type `Nothing` is polykinded type, so it can also be
+  * applied as an argument to the type parameter `F[_]`.
+  * Nevertheless, for the sake of clarity, in `fs2` we use the type alias
+  * `type Pure[A] = Nothing` to represent a pure pull as `Pull[Pure, O, R]`.
+  *
+  * A '''silent''' pull is one known to emit no outputs. In a silent pull, the
```

Comment:
Not sure I want to introduce this silent, broken, and closed terminology -- three new concepts which we don't currently use anywhere else.

Reply:
There are new concepts/words. The reason is, a lot of places had the words "that returns unit" or "without effects", so I thought that a quick word to indicate that would be helpful.

It also helps later on the method or constructors below, like "fail" or "pure".

Reply:
Yeah, I would much prefer we just let the types do the talking here. I find these new concepts *harder* to understand than just looking at the types. For example:
- Why is a pull which has the output type = `Nothing` considered silent but a pull which has `O != Nothing` and yet outputs no elements not silent?
- If a pull with `O = Nothing` is silent, are pulls with `O != Nothing` audible? Loud?
- Are broken pipes fixable?
- Are pipes with `R != Unit` open?
I find these analogies more confusing than helpful.

Reply:
Ok, I have removed these words and added some notes to the type parameter descriptions.